### PR TITLE
:sparkles: Deliver operator for status via API

### PR DESCRIPTION
**Pull request by jheubuch**
With this PR, the HAFAS operator is delivered within the StatusResource.

### CODE REVIEW

File: app/Http/Resources/StatusResource.php [hunk 1]
```diff
@@ -48,6 +48,10 @@ public function toArray($request): array {
                 'manualArrival'       => $this->trainCheckin->manual_arrival?->toIso8601String(),
                 'origin'              => new StopoverResource($this->trainCheckin->origin_stopover),
                 'destination'         => new StopoverResource($this->trainCheckin->destination_stopover),
+                'operator'            => [
```
Comment (HerrLevin) [inline app/Http/Resources/StatusResource.php]: Please convert this to a nullable OperatorResource.

File: app/Virtual/Models/Train.php [hunk 2]
```diff
@@ -142,4 +142,14 @@ class Train
...
      */
     private $destination;
+
+    /**
+     * @OA\Property (
+     *     title="operator",
+     *     description="Operator of the mean of transport"
+     * )
+     *
+     * @var Operator
```
Comment (HerrLevin) [inline app/Virtual/Models/Train.php:153]: this needs to be nullable

**Review by HerrLevin — CHANGES_REQUESTED** [review]

File: app/Http/Resources/StatusResource.php (hunk 1 above)
Comment (jheubuch) [reply app/Http/Resources/StatusResource.php]: How do I do that? 🙃 (I have no idea using PHP)

Comment (jheubuch) [reply app/Http/Resources/StatusResource.php]: Ok, I tried and tested but maybe there is a better way

Comment (HerrLevin) [reply app/Http/Resources/StatusResource.php]: You can create resources in Laravel with `php artisan make:resource OperatorResource`.

Then just have a look at the implementation of StopoverResource. :)

Comment (jheubuch) [reply app/Http/Resources/StatusResource.php]: Ok I guess this is the better way than using `is_null(..) ? null : [...]` :D

Comment (HerrLevin) [reply app/Http/Resources/StatusResource.php]: Exactly. :D

File: app/Http/Resources/OperatorResource.php [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+     * Transform the resource into an array.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return array|\Illuminate\Contracts\Support\Arrayable|\JsonSerializable
+     */
+    public function toArray($request)
+    {
+        return [
+            'id'    => $this->hafas_id,
+            'name'  => $this->name
```
Comment (HerrLevin) [inline app/Http/Resources/OperatorResource.php]:
```suggestion
            'identifier'    => $this?->hafas_id,
            'name'  => $this?->name
```

File: app/Virtual/Models/Operator.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+class Operator
+{
+    /**
+     * @OA\Property (
+     *     title="id",
+     *     example="sbb"
+     * )
+     * @var string
+     */
+    private $id;
```
Comment (HerrLevin) [inline app/Virtual/Models/Operator.php]:
```suggestion
    /**
     * @OA\Property (
     *     title="identifier",
     *     example="sbb"
     * )
     * @var string
     */
    private $identifier;
```

File: app/Http/Resources/StatusResource.php [hunk 5]
```diff
@@ -48,6 +48,7 @@ public function toArray($request): array {
                 'manualArrival'       => $this->trainCheckin->manual_arrival?->toIso8601String(),
                 'origin'              => new StopoverResource($this->trainCheckin->origin_stopover),
                 'destination'         => new StopoverResource($this->trainCheckin->destination_stopover),
+                'operator'            => new OperatorResource($this?->trainCheckin->operator)
```
Comment (HerrLevin) [inline app/Http/Resources/StatusResource.php]:
```suggestion
                'operator'            => $this->trainCheckin->HafasTrip->operator ? new OperatorResource($this->trainCheckin->HafasTrip->operator) : null
```

**Review by HerrLevin — CHANGES_REQUESTED** [review]

**Pull request merged**